File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/AccountTargetProperty.java
package org.pmiops.workbench.actionaudit.targetproperties;

import jakarta.validation.constraints.NotNull;

public enum AccountTargetProperty implements SimpleTargetProperty {
  IS_ENABLED("is_enabled"),
  ACKNOWLEDGED_TOS_VERSION("acknowledged_tos_version"),
  INITIAL_CREDITS_OVERRIDE("initial_credits_override"),
  ACCESS_TIERS("access_tiers");

  private final String propertyName;

  AccountTargetProperty(String propertyName) {
    this.propertyName = propertyName;
  }

  @NotNull
  @Override
  public String getPropertyName() {
    return propertyName;
  }
}

File: api/src/main/java/org/pmiops/workbench/actionaudit/targetproperties/AccountTargetProperty.kt
package org.pmiops.workbench.actionaudit.targetproperties

enum class AccountTargetProperty
    constructor(override val propertyName: String) : SimpleTargetProperty {
    IS_ENABLED("is_enabled"),
    REGISTRATION_STATUS("registration_status"),
    ACKNOWLEDGED_TOS_VERSION("acknowledged_tos_version"),
    FREE_TIER_DOLLAR_QUOTA("free_tier_dollar_quota"),
    ACCESS_TIERS("access_tiers");
}
